haskell-src/Concordium/Genesis/Data/P6.hs

```diff
@@ -30,8 +74,16 @@ data GenesisDataP6 = GDP6Initial
 -- The relationship between the new state and the state of the
 -- terminal block of the old chain should be defined by the
 -- chain update mechanism used.
-newtype RegenesisP6 = GDP6Regenesis {genesisRegenesis :: BaseV1.RegenesisDataV1}
-    deriving (Eq, Show)
+--
+-- There are two variants, one when migrating from 'P5' to 'P6'and
+-- one from 'P6' to 'P6'.
+data RegenesisP6
+    = GDP6Regenesis {genesisRegenesis :: !BaseV1.RegenesisDataV1}
+    | GDP6RegenesisFromP5
+        { genesisRegenesis :: !BaseV1.RegenesisDataV1,
+          genesisMigration :: !StateMigrationData
+        }
+    deriving stock (Eq, Show)
```

Comment:
I don't think it is useful to specify the `stock` deriving strategy here, or in general.